archethic/contracts/factory.exs

```diff
@@ -234,14 +249,15 @@ export fun get_signed_htlc(user_address, pool_address, token, amount) do
     @version 1
     # Automate the refunding after the given timestamp
     actions triggered_by: datetime, at: \#{end_time} do
-    #{refund_code}
+      htlc_genesis_address = Chain.get_genesis_address(contract.address)
+      Contract.add_recipient(address: 0x#{pool_address}, action: "refund", args: [htlc_genesis_address])
     end
 
-    condition triggered_by: transaction, on: refund(), as: [
+    condition triggered_by: transaction, on: refund(secret, secret_signature), as: [
```

Comment:
You should ensure this trigger is coming from the pool, otherwise anybody could refund using invalid secret and signature